files/common/usr/bin/delphix-startup-screen

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/python3
+#
+# Copyright 2020 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import curses
+import curses.ascii
+import curses.textpad
+import sys
+import os
+import logging
+import time
+import subprocess
+
+STATUS_HEADER = "STATUS\t\t\tSERVICE\t\t\t"
+LAYOUT_STR = "Keyboard layout (press 9 to change):"
+CONSOLE_STR = "Press any other key for console access."
+KEY_LAYOUT_SELECT_STR = "Select a keyboard layout by number [1-%d]:"
+
+log_file = "/tmp/dpx_startup_screen.log"
+logo = "Delphix Engine Version "
+
+website = "www.delphix.com"
+
+WIN_LEN = 0
+WIN_HT = 0
+
+def get_svcs():
+    cmd = ['systemctl', 'show', '-p',  'Wants', 'delphix.target']
+    cp = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True)
+    return cp.stdout.split("=", 1)[1].split()
+
+def run_svcs(options=None):
+    '''
+    Run the command to retrieve the status of all the services. The options
+    argument can be used to obtain additional output from the 'systemctl'
+    command. By default we only obtain the 'ActiveState' of the services.
+    '''
+
+    fields = 'ActiveState'
+    cmd = ['systemctl', 'show', '--value', '-p']
+    if options:
+        fields += ',' + options
+    cmd.append(fields)
+    for svc in get_svcs():
+        cmd.append(svc)
+    cp = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True)
+    return cp.stdout
+
+def get_keyboard_layout():
+    cmd = ['localectl', 'status']
+    cp = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True)
+    output = cp.stdout.split()
+    return output[output.index("Layout:") + 1]
+
+def set_keyboard_layout(layout):
+    cmd = ['localectl', 'set-x11-keymap', layout, 'pc105']
+    cp = subprocess.run(cmd, shell=False)
+    return subprocess.run('setupcon', shell=False)
+
+def get_valid_keyboard_layouts():
+    cmd = ['localectl', '--no-pager', 'list-x11-keymap-layouts']
+    cp = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True)
+    return cp.stdout.splitlines()
+
+def load_header(stdscr):
+    global logo
+
+    cmd = ['/opt/delphix/server/bin/get_appliance_version']
+    cp = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True)
+    version = cp.stdout.split('-', 1)[0]
```

Comment:
I don't feel strongly, but we could potentially add an option to the `get-appliance-version` script to break up the version information like you're doing here.. this way, that same functionality could be use elsewhere, rather than forcing the consumers to know how to properly split up the string that is returned.

e.g. something like a `--major` and/or `--minor` options.. or maybe allow a printf style string to be supplied.. 

With that said, I'm good with what you have as-is.

Reply:
What do you think about making the default always report the first part of the version and having `--full` or `--debug` to get the snapshot info?

Reply:
Seems reasonable; I'm not sure how that'll interact with other software using that script, but seems reasonable to me at first thought.

Reply:
I looked at the other consumers of `get-appliance-version` in `delphix-platform` repo and I think your original proposal of `--major` made more sense. I was trying to determine the use cases for `--minor` and `--patch`. Right now they return back just that portion of the version but I'm not sure how useful that would be especially since I'm not currently restricting the options to be mutually exclusive. If I make the options mutually exclusive then we can either have `--minor` return back just the minor portion of the version or return back the whole version minus the patch portion. In the latter case, it makes the need for `--patch` unnecessary since it would overlap with the default case. Let me know what you think.

Reply:
> If I make the options mutually exclusive then we can either have --minor return back just the minor portion of the version or return back the whole version minus the patch portion.

Yea, I think it makes sense to make them exclusive, and return the version up to the option specified; i.e. major is just the two major number, minor is the 3 numbers, patch is the 4 numbers, etc.

> In the latter case, it makes the need for --patch unnecessary since it would overlap with the default case. 

Would `--patch` also return the snapshot information? e.g. on my system:
```
6.1.0.0-snapshot.20200325201630621+jenkins-ops-devops-gate-master-appliance-build-master-post-push-2822
```
I could imagine `--patch` only returning `6.0.0.0`

I could also imagine a `--snapshot` option that returns: `6.1.0.0-snapshot.20200325201630621` (this would be used by the MOTD script)

and then the default, that'd return the entire string.

> I looked at the other consumers of get-appliance-version in delphix-platform repo

There's also "consumers" in the app-gate.. those are currently using the app-gate's `get_appliance_version` script, but arguably those should be using the platform script.. e.g. if we change how the version is stored, we'd update the platform script, and the app-gate would not have to know about that.

But I think all the consumers of the app-gate would be able to use the new options that you mentioned (e.g. `--major`, `--minor`, etc.).

Reply:
Here's the output of all the options:
```
# ./get-appliance-version 
6.1.0.0
# ./get-appliance-version --patch
6.1.0.0
# ./get-appliance-version --snapshot
6.1.0.0-snapshot.20200325231145826
# ./get-appliance-version --major
6.1
# ./get-appliance-version --minor
6.1.0
# ./get-appliance-version --debug
6.1.0.0-snapshot.20200325231145826+jenkins-ops-devops-gate-master-appliance-build-master-post-push-2824
```